util/topgen/validate.py

```diff
@@ -757,9 +757,12 @@ def validate_clock(top, inst, clock_srcs, prefix=""):
                   (prefix, name))
         [log.error("%s" % port) for port in missing_port]
 
-    missing_net = [
-        net for port, net in top['clock_srcs'].items() if net not in clock_srcs
-    ]
+    missing_net = []
+    for port, net in top['clock_srcs'].items():
+        net = net['clock'] if isinstance(net, Dict) else net
```

Comment:
As above, I'd consider calling this variable something like `clock_name` rather than changing the type of `net`.

Reply:
updated.